---
output: 
html_document:
  number_sections=TRUE
pagetitle: Assignment_5
---
    
# **Bioinformatics Data Skills**
    
## Assignment 5

___

  **1. Read through Chapter 7 at least twice**
    
  **2. It's a big chapter which is why we've got 2 weeks with it, but pay attention to things like grep, sed, awk, etc... **
    
  **3. Here's a small [fasta file](https://gzahn.github.io/binf-data-skills/Data/Assignment_5_File_1.fasta) you should download into an appropriate place in your Assignment_5 project directory.**
        
  **4.  I need you to convert that file into tabular format using some combination of the tools you've learned. Tabular format looks like the following:**
  
    OTU8      TACGGAGGGTGCAAGC...
    OTU11     TGGCAGACTAGAGTAT...
 
  Note: I want all the descriptive taxonomic stuff removed. The first column is *just* the OTUID, the second column is *just* the sequence          
  
  **5.  Keep track of any steps you take to convert from fasta to that tabular (with two columns) format, and save the new version of the data file as "Assignment_5_File_1.tsv**
  
  **6.  Now, starting from that new tabular file, use awk/sed/etc. to convert it back to standard fasta format. Record all your commands that got the job done.**
  
  **7.  Okay, back to the orginal fasta file...two of the DNA reads have been assigned the exact same taxonomy. Write a short command to find the full taxonomic affiliation of the repeated taxonomic group. Your command can't make use of the taxonomic names in any way; Just because you can visually inspect this file doesn't mean it's always possible or feasible.**
  
  **8.  Now, extend that command to return _just_ the two DNA reads where the taxonomy is the same. Let me show you want I mean:**
  
    >OTU15_Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Psychrobacter
    TACAGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGAGCGTAGGTGGCTTGATAAGTCAGATGTGAAATCCCCGGGCTTAACCTGGGAACTGCATCTGATACTGTCAGGCTAGAGTAGGTGAGAGGAAGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGATGGCGAAGGCAGCCTTCTGGCATCATACTGACACTGAGGCTCGAAAGCGTGGGTAGCAAACAGG
    >OTU28_Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Psychrobacter
    TACAGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGAGCGTAGGTGGCTTAATAAGTCAGATGTGAAATCCCCGGGCTTAACCTGGGAACTGCATCTGATACTGTTGGGCTAGAGTAGGTGAGAGGGAGGTAGAATTTCAGGTGTAGCGGTGAAATGCGTAGAGATCTGAAGGAATACCGATGGCGAAGGCAGCCTCCTGGCATCATACTGACACTGAGGTTCGAAAGCGTGGGTAGCAAACAGG

So, in other words, your full command should search the file for matching taxonomy, and then use that match to just show the fasta records for those two matching reads. Note that this is in the "proper" fasta format, with each record taking up exactly 2 lines.

  **9.  This whole process may look short and sweet, but it will require sed, grep, awk, paste, process substitution, and some other stuff, all rolled into one!**
  
___
  
### **What to turn in:**

  - Store a cleaned-up plain-text file that documents the code you used for tasks 4-8
  - Be sure to include comments explaining what is going on in your code
  
Upload that plain-text file to Canvas. You'll be graded on completeness, documentation, and readability.

___

<br/>