---
output: 
html_document:
  number_sections=TRUE
pagetitle: Case_Study_1
---
    
# **Bioinformatics Data Skills**
    
## Case Study 1: Use an existing tool to accompish a customized task

<br/><br/><br/><br/>


<div style= "float:right;position: relative;top:10px">
```{r, out.width = "350px",echo=FALSE}
knitr::include_graphics("../../Media/ITS.png")
```
</div>

___


  +  **The internally transcribed spacer region is a region between highly conserved the small subunit (SSU) of rRNA and the large subunit (LSU) of the rRNA. **
  
  +  **In Eukaryotes it contains the 5.8s genes and two variable length spacer regions.** 
  
  +  **In amplicon sequencing studies it is common practice to trim off the conserved (SSU, 5,8S or LSU) regions.  **
  
  +  **The conserved sites, including where the primers sit, can mess up taxonomic assignment algorithms, and is generally not informative.**

___
  
#### So, we're going to be using an existing piece of software to extract just the ITS1 region from a series of DNA reads.  The tool we will use is called [itsxpress](https://github.com/usda-ars-gbru/itsxpress){target="_blank"} 

  +  **Getting it installed and working properly is the first task:**

  - It depends on a lot of other software, including [HMMER3](http://hmmer.org/){target="_blank"}, [biopython](https://biopython.org/wiki/Download){target="_blank"}, and others
  - You will need to install the [conda](https://docs.conda.io/en/latest/miniconda.html){target="_blank"} system in order to even install itsxpress
  - [miniconda installation instructions](https://conda.io/projects/conda/en/latest/user-guide/install/index.html){target="_blank"}
    
  +  **Once you have installed itsxpress and its dependencies, make sure it's in your $PATH so the program can be found, and then take a look at how to use it from the help page.**
  
  +  **We need to run itsxpress on all the fastq files in our [data set](https://gzahn.github.io/binf-data-skills/Case_Studies/01_extract_ITS1_region/data/ITS_Reads.zip), making sure to stash all the output information and diagnostics for each file.**
   
  That data set, once unzipped, contains 15 samples (each in its own file) of ITS amplicons. The sample names are:
  
        CC1,CC2,CC3,CC4,CC5,CR1,CR2,CR3,CR4,CR5,PF1,PF3,PF5,SW1,SW2
  The problem is that the amplicons have overhanging regions outside of the ITS1 that we want, so wee need to extract *just* the regions of reads in each sample that corresponds to the ITS1 section and remove the partial 18S and 5.8S sections.
  
  +  **In other words, our script should automatically perform the following for each file:**
  
    1. Extract the ITS1 regions from each sample (AKA, from each file)
    2. Non-default parameters for ITSx should be:
      - suppress 'Graphical' output
      - preserve sequence headers
      - save *only* the ITS1 region
      - base name for output should be "ITS1"
      - do *not* check the reverse compliment of the reads
    3. Store all diagnostic log information in a unique file for each sample
  
  +  **Further, we want to get some summary information about how many reads in each file successfully had an ITS1 detection and extraction**
  
  +  **Use grep, sed, awk, cut, paste (or whatever tools you need) to extract summary info out of the log files about how well itsxpress performed**
  
    1. How many input sequences were there?
    2. How many output sequences made it through?
    3. Save this selected summary info (for each sample) into a new file called "ITSxpress_summary_info.txt"
      - This file should be organized well enough so that for each sample, it's easy to see the selected summary info associated with it... something like this:

      CC2
      Number of sequences in input file:       	6791
      Sequences detected as ITS by ITSx:	6520
      PF1
      Number of sequences in input file:       	6141
      Sequences detected as ITS by ITSx:	4845

    
  
___


### **What to turn in:**

  - Store a cleaned-up plain-text file that documents the code you used to accomplish everything
  - Be sure to include comments explaining what is going on in your code
  - Upload that plain-text file to Canvas. You'll be graded on completeness, documentation, and readability.
  - Copy/Paste the contents of ITSxpress_summary_info.txt into the text entry box of the Canvas assignment as well.
  


___

<br/>